worker/meterstatus/manifold.go

```diff
@@ -94,11 +100,34 @@ func newStatusWorker(config ManifoldConfig, context dependency.Context) (worker.
 	}
 	logger.Tracef("Starting connected meter status worker.")
 	status := config.NewMeterStatusAPIClient(apiCaller, unitTag)
+	stateReadWriter := NewControllerBackedState(
+		config.NewUniterStateAPIClient(
+			base.NewFacadeCaller(apiCaller, "MeterStatus"),
+			unitTag,
+		),
+	)
+
+	// Check if a local state file exists from a previous isolated worker
+	// instance. If one is found, migrate it to the controller and remove
+	// it from disk; this doubles as an auto-magic migration step.
+	if priorState, err := NewDiskBackedState(localStateFile).Read(); err == nil {
```

Comment:
If `Read` fails then it will continue, even if a `Read` fails on a yaml file. I feel like we need to bubble up the error or log it if the yaml has an error in it.

Reply:
You are right. It should log non-NotFound errors.